sql/src/main/java/io/crate/executor/transport/task/elasticsearch/EsJobContextTask.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to CRATE.IO GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.executor.transport.task.elasticsearch;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import io.crate.executor.JobTask;
+import io.crate.executor.TaskResult;
+import io.crate.jobs.ESJobContext;
+import io.crate.jobs.JobContextService;
+import io.crate.jobs.JobExecutionContext;
+import io.crate.operation.projectors.FlatProjectorChain;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.support.TransportAction;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.UUID;
+
+public class EsJobContextTask extends JobTask {
+
+    protected final List<SettableFuture<TaskResult>> results;
+    protected final int executionPhaseId;
+    private final JobContextService jobContextService;
+    protected JobExecutionContext context;
```

Comment:
no unfortunately it cannot created at ctor (will be created due to its builder), there is a dedicated createContext method for it